### PR TITLE
test: add help text test

### PR DESCRIPTION
**Pull request by yvmunayev**
fix: #1607 

## Changes proposed in this PR:
- test: add help text test 
- refactor: refactor defaultPositionResolver on InternalOverlay

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

**Comment by TahimiLeonBravo** [comment]
@yvmunayev fix positionResolver
Screen Shot 2020-06-23 at 9 46 56 PM

**Comment by TahimiLeonBravo** [comment]
@yvmunayev add box-shadow styles when the button id focused

**Comment by TahimiLeonBravo** [comment]
@yvmunayev the HelpText message should close when the window resize

https://drive.google.com/file/d/1FisnzjtXsylVphZBOBUdoaOdrcNz4cSz/view?usp=sharing

File: src/components/HelpText/__test__/helpText.spec.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        wrapper.find('button').simulate('blur');
+        expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(false);
+    });
+
+    it('should show info when mouse enter to button and hidden when leave', () => {
+        const wrapper = mount(<HelpText text="Help Text" />);
+        expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(false);
+        wrapper.find('button').simulate('mouseenter');
+        expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(true);
+        wrapper.find('button').simulate('mouseleave');
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/__test__/helpText.spec.js:30]: same ^^

File: src/components/HelpText/__test__/helpText.spec.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        setTimeout(() => {
+            expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(false);
+        });
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/__test__/helpText.spec.js:33]: can we avoid this setTimeout

File: src/components/HelpText/__test__/helpText.spec.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        const wrapper = mount(<HelpText text="Help Text" />);
+        wrapper.find('button').simulate('focus');
+        expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(true);
+        wrapper.find('button').simulate('mouseleave');
+        setTimeout(() => {
+            expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(true);
+        });
+    });
+
+    it('should maitime focus bottom when clicking in help text ', () => {
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/__test__/helpText.spec.js:56]: maitime?

File: src/components/HelpText/helpers/positionResolver.js [hunk 4]
```diff
@@ -1,60 +1,18 @@
...
-            top: trigger.leftUpAnchor.y - content.height,
-            left: trigger.rightUpAnchor.x - content.width,
-        };
-    }
-    return false;
-};
+import resolverBottomLeft from '../../InternalOverlay/helpers/resolverBottomLeft';
+import resolverBottomRight from '../../InternalOverlay/helpers/resolverBottomRight';
+import resolverUpLeft from '../../InternalOverlay/helpers/resolverUpLeft';
+import resolverUpRight from '../../InternalOverlay/helpers/resolverUpRight';
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/helpers/positionResolver.js:4]: we need to discuss about this additions to InternalOverlay

File: src/components/HelpText/helpers/resolverBottomCenter.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export default function resolverBottomCenter(opts) {
+    const { trigger, content, viewport } = opts;
+    if (trigger.leftBottomAnchor.y + content.height <= viewport.height) {
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/helpers/resolverBottomCenter.js:3]: these resolver should be part of the internal overlay

File: src/components/HelpText/helpers/positionResolver.js (hunk 4 above)
Comment (TahimiLeonBravo) [reply src/components/HelpText/helpers/positionResolver.js:4]: @yvmunayev I think also we should add resolver for rightCenter and leftCenter positions

File: src/components/HelpText/helpers/positionResolver.js [hunk 6]
```diff
@@ -1,60 +1,22 @@
...
+const resolvers = [
+    resolverBottomRight,
+    resolverBottomLeft,
+    resolverBottomCenter,
+    resolverUpRight,
+    resolverUpLeft,
+    resolverUpCenter,
+    resolverCenterRight,
+    resolverCenterLeft,
+];
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/helpers/positionResolver.js:19]: The internal overlay default resolver should use this by default, there is no need to repeat this here.
What we want is that in the great majority use cases of the InternalOverlay the consumer does not need to use the position resolver, but only for exceptional cases.

File: src/components/HelpText/styled/index.js [hunk 7]
```diff
@@ -1,10 +1,11 @@
 import styled from 'styled-components';
 import attachThemeAttrs from '../../../styles/helpers/attachThemeAttrs';
 
-export const StyledButton = styled.button`
+export const StyledButton = attachThemeAttrs(styled.button)`
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/styled/index.js:4]: remove attachThemeAttrs if it is not used

File: src/components/HelpText/helpers/positionResolver.js (hunk 6 above)
Comment (yvmunayev) [reply src/components/HelpText/helpers/positionResolver.js:19]: In the component design it has 2 differences: first the margin is 0 and second the order of is different.

Comment (LeandroTorresSicilia) [reply src/components/HelpText/helpers/positionResolver.js:19]: @yvmunayev Design is only a representation and this is meant to not be used here in this component

File: src/components/HelpText/__test__/helpText.spec.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    it('should show info when button is focused and hiddden when is not', () => {
+        const wrapper = mount(<HelpText text="Help Text" />);
+        expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(false);
+        wrapper.find('button').simulate('focus');
+        expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(true);
+        wrapper.find('button').simulate('blur');
+        expect(wrapper.find(InternalOverlay).prop('isVisible')).toBe(false);
+    });
+
+    it.skin('should show info when mouse enter to button and hidden when leave', () => {
```
Comment (LeandroTorresSicilia) [inline src/components/HelpText/__test__/helpText.spec.js:25]: skin?

**Pull request merged**